app/src/main/java/com/arctouch/codechallenge/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,81 @@
+package com.arctouch.codechallenge.ui.home
+
+import android.arch.lifecycle.MutableLiveData
+import android.view.View
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.api.TmdbApi
+import com.arctouch.codechallenge.base.BaseViewModel
+import com.arctouch.codechallenge.data.Cache
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.util.API_KEY
+import com.arctouch.codechallenge.util.DEFAULT_LANGUAGE
+import com.arctouch.codechallenge.util.DEFAULT_REGION
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import javax.inject.Inject
+
+
+class HomeViewModel : BaseViewModel() {
+    @Inject
+    lateinit var tmdbApi: TmdbApi
+    private lateinit var subscription: Disposable
+    val movieListAdapter: HomeAdapter = HomeAdapter()
+
+    val loadingVisibility: MutableLiveData<Int> = MutableLiveData()
+
+    val totalPages: MutableLiveData<Int> = MutableLiveData()
+
+    val errorMessage: MutableLiveData<Int> = MutableLiveData()
+
+    init {
+        getGenres()
+        getUpcomingMovies()
```

Comment:
Which of these API calls (not the method itself, but the network call that they trigger) finishes first? Does the order matter?

Reply:
In order: first getGenres then getUpcomingMovies. The subscription variable will work individually for each function.